### PR TITLE
CLN: remove align kwarg from Block.where, Block.putmask

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -406,13 +406,13 @@ def apply(self: T, f, filter=None, **kwargs) -> T:
 
         if f == "where":
             align_copy = True
-            if kwargs.get("align", True):
+            if kwargs.pop("align", True):
```
Comment (WillAyd) [inline pandas/core/internals/managers.py]: Is this necessary? I think better to not mutate the argument, unless we have full control over where those arguments are coming from

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py]: We futz with kwargs down on L442 anyway.  We could copy the dict and mess with that, but that doesnt seem better to me

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -406,13 +406,13 @@ def apply(self: T, f, filter=None, **kwargs) -> T:
...
                 align_keys = ["other", "cond"]
             else:
                 align_keys = ["cond"]
         elif f == "putmask":
             align_copy = False
-            if kwargs.get("align", True):
+            if kwargs.pop("align", True):
```
Comment (jreback) [inline pandas/core/internals/managers.py]: can you completely remove align?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/managers.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py]: id like to, but havent been able to yet

Comment (jreback) [reply pandas/core/internals/managers.py]: kk

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py]: in an upcoming PR im going to move the defining of align_keys to the calling function, so "align" can be popped there

Comment (jbrockmendel) [reply pandas/core/internals/managers.py]: xref #32846

**Comment by jreback** [comment]
I thought we merged this, pls rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**